src/resolver/mod.rs

```diff
@@ -77,16 +79,16 @@ impl Stream for Resolve {
 /// Serves resolutions from `Resolver`s.
 pub struct Executor {
     requests: mpsc::UnboundedReceiver<(Path, mpsc::UnboundedSender<Result<Vec<WeightedAddr>>>)>,
-    namerd: Namerd,
+    namer: Box<Namer + Send>,
```

Comment:
I think if you always want the `Send` bound on `Namer`, you might want to consider making it part of the trait, instead:
```rust
pub trait Namer: Send {
    ...
}
```